scripts/check_libc_dependencies.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+ROOT_SRC_DIR="$SCRIPT_DIR/.."
+
+if [[ ! -f "$ROOT_SRC_DIR/CMakeLists.txt" ]]; then
+    echo "Failed to find note-c root directory. (did the location of check_libc_dependencies.sh change?)"
+    exit 1
+fi
+
+pushd $ROOT_SRC_DIR $@ > /dev/null
+
+CMAKE_OPTIONS="-DNOTE_C_NO_LIBC=1"
+
+cmake -B build/ $CMAKE_OPTIONS
+if [[ $? -ne 0 ]]; then
+    echo "Failed to run CMake."
+    popd $@ > /dev/null
+    exit 1
+fi
+
+PERMITTED_FNS=(
+    # The mem* functions are ok.
+    "memchr"
+    "memcmp"
+    "memcpy"
+    "memmove"
+    "memset"
+    # These string functions are ok.
+    "strchr"
+    "strcmp"
```

Comment:
This seems like a good opportunity to not allow these unsafe functions.

Reply:
FWIW, I don't think this should prevent this commit from happening, but we should add it to the backlog - especially now that there is a mechanism for protecting against this.

Reply:
I'd have to re-immerse myself in the discourse on this age-old argument, but my rough understanding is that you're not that much better off using strncmp over strcmp: https://stackoverflow.com/questions/24353504/whats-wrong-with-strcmp

If your string isn't NULL-terminated, all bets are off.